README.md

```diff
@@ -1,2 +1,40 @@
-# rik
+# RIK
+
 A rustlang based cloud orchestrator
+
+## RIK SDN
```

Comment:
My proposal for an incremental design, based on the assumption that  we will schedule containers (or pods) with their own networking namespace (one netns per container or per pod):

## v1

* All containers on a node are connected through a single Linux bridge
* All containers on the cluster are part of the same subnet, a /16 for example
* The initial SDN should be a per node daemon that simply
    a. Takes a netns (The one created by the node agent when creating the container) and an IP from the global subnet (allocated by the controller).
    b. Creates a veth pair with one end attached to the node Linux bridge, and another end attached to the container netns
   c. Sets the  container end of the veth pair with the provided IP.

## v2

* Add external connectivity to the created bridge (NAT+ipv4 forwarding) and allow for all containers to be externally accessible

## v3

* One subnet per tenant
* One bridge per tenant on each node

## v4

* Overlay networking: Link all tenant bridges together, across the cluster


We should initially focus on v1, and provide a simple gRPC API for the node agent to send a networking setup request that would take a netns and an IP address, and that would set the container netns up and link it to the host local Linux bridge.

Some references:

- https://www.cni.dev/plugins/current/main/bridge/
- https://karampok.me/posts/container-networking-with-cni/